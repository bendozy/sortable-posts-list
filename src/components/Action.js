import React from 'react';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';
import timeTravel from '../actions/actions';

export const Action = ({
  action: { title, from, to },
  timeTravel: timeTravelAction,
  index,
}) => (
  <div className="Action bg-white rounded p-3 shadow border flex flex-wrap">
    <div className="Action-message sm:my-3 my-1 sm:text-base text-sm w-3/4 text-gray-700">
      {`Moved ${title} from index ${from} to index ${to}`}
    </div>
    <button
      onClick={() => timeTravelAction(index)}
      type="button"
      className="Action-button h-12 m-1 bg-green-200 rounded text-sm shadow sm:text-base w-1/5 text-gray-700"
    >
      Time Travel
    </button>
  </div>
);

Action.propTypes = {
  action: PropTypes.shape({
    title: PropTypes.string.isRequired,
    from: PropTypes.number.isRequired,
    to: PropTypes.number.isRequired,
  }).isRequired,
  index: PropTypes.number.isRequired,
  timeTravel: PropTypes.func.isRequired,
};

export const mapDispatchToProps = {
  timeTravel,
};

export default connect(null, mapDispatchToProps)(Action);
