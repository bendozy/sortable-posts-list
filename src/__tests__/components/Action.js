import React from 'react';
import toJson from 'enzyme-to-json';
import Action from '../../components/Action';

describe('<Action />', () => {
  let wrapper;
  const timeTravel = jest.fn();

  beforeEach(() => {
    const action = {
      title: 'test',
      from: 0,
      to: 1,
    };
    wrapper = shallow(
      <Action timeTravel={timeTravel} action={action} index={0} />,
    );

    wrapper.find('.Action-button').simulate('click');
  });

  it('should render a <Action /> component', () => {
    expect(wrapper.find('.Action')).toHaveLength(1);
    expect(wrapper.find('.Action-message')).toHaveLength(1);
    expect(wrapper.find('.Action-message').text()).toBe('Moved test from index 0 to index 1');
    expect(wrapper.find('.Action-button').text()).toBe('Time Travel');
    expect(timeTravel.mock.calls.length).toBe(1);
    expect(toJson(wrapper)).toMatchSnapshot();
  });
});
