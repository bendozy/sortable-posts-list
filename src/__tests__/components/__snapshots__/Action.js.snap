// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Action /> should render a <Action /> component 1`] = `
<div
  className="Action bg-white rounded p-3 shadow border flex flex-wrap"
>
  <div
    className="Action-message sm:my-3 my-1 sm:text-base text-sm w-3/4 text-gray-700"
  >
    Moved test from index 0 to index 1
  </div>
  <button
    className="Action-button h-12 m-1 bg-green-200 rounded text-sm shadow sm:text-base w-1/5 text-gray-700"
    onClick={[Function]}
    type="button"
  >
    Time Travel
  </button>
</div>
`;
